### PR TITLE
Tidy up pearl output files naming and save directories

### LINKED ISSUE #30883: PEARL output file requests - Nice to have
**Original reporter:** Nick Funnell

- 1) With `focus_mode=‘All’`, all file names are appended with a “-0”. The -0 signifies merged banks 1-9, so perhaps the -0 could be dropped when it’s a single bank, e.g. `XXXXXX_tof_xye-10-0.dat`
- 2) Also could the `_xye` bit be removed and `.dat’` replaced with `.xye`. e.g. e.g. XXXXXX_tof_xye-10-0.dat -> e.g. `XXXXXX_tof-10-0.xye`. This would mirror the filename behaviour of `focus_mode=“Trans”`.
-  3) At the moment, all our focussed files go into a single folder. It would be nice if we could separate these into sub-folders named by file type. I’m envisaging folders for ‘ToF’, ‘d’, and ‘GSAS’ and then the .nxs files would continue to just go into the current directory as normal. I think it’s done on some of the other instruments.

### PR DESCRIPTION
**Pull request by RichardWaiteSTFC**
**Description of work.**

Tidy up pearl output files naming and save directories

**To test:**

Follow Danny's instructions on a comment on the original issue:

> Script to run a PEARL reduction. The C:\PEARL\ folder will need changing to wherever your config files are stored:
> 
> ```
> # import mantid algorithms, numpy and matplotlib
> from mantid.simpleapi import *
> import matplotlib.pyplot as plt
> import numpy as np
> 
> from isis_powder.pearl import Pearl
> 
> pearl = Pearl(user_name="Danny",
>                     output_directory=r"C:\PEARL",   #directory above cycle number 
>                     config_file=r"C:\PEARL\User_experiment.yaml",
>                     calibration_directory=r"C:\PEARL\calib_critical_files",
>                     calibration_mapping_file=r"C:\PEARL\calib_critical_files\pearl_run_mapping.yaml")
> 
> pearl.create_vanadium(run_in_cycle=116374, do_absorb_corrections=True)
> 
> pearl.focus(run_number="116374", file_ext=".s02", perform_attenuation=False, 
>                   subtract_empty_instrument=False, focus_mode="all")
> ```
> 
> In addition to the script above you'll also need a zip file with some config (.yaml) files in it and a couple of data files (eg a .nxs file). This is too big to attach to this issue so I'll put it in \\olympic\Babylon5\Public\DannyHindson\PEARLExampleForSarah.zip

There should be separate directories for tof and d-spacing xye files and the .gss files
image
The .nxs files are still saved in the top directory as before.

Fixes #30883 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Diffraction/isis_powder/pearl.py [hunk 1]
```diff
@@ -89,9 +89,46 @@ def create_cal(self, **kwargs):
...
+            "suffix": run_details.output_suffix if run_details.output_suffix else "",
+        }
+        format_options = self._add_formatting_options(format_options)
+
+        output_formats = {
+            "nxs_filename": output_directory,
+            "gss_filename": os.path.join(output_directory, "GSAS"),
+            "tof_xye_filename": os.path.join(xye_files_directory, "ToF"),
+            "dspacing_xye_filename": os.path.join(xye_files_directory, "dSpacing"),
+        }
```
Comment (DannyHindson) [inline scripts/Diffraction/isis_powder/pearl.py:122]: This Pearl-specific function override of `_generate_out_file_paths` copies all of the code from the equivalent function in abstract_inst.py but I think the only changes are L119-121. The `dat_files_directory` variable is also renamed higher up but that's not important. Please could you create a new function in abstract_inst.py called `_get_output_formats()` or similar that generates the output_formats dictionary
and then have an overridden version of `_get_output_formats` in this unit that does the Pearl-specific behaviour (subfolders)? You can then remove the Pearl specific version of `_generate_out_file_paths` I think

**Review by DannyHindson — CHANGES_REQUESTED** [review]

**Review by DannyHindson — DISMISSED** [review]
Looks good thanks

**Comment by DannyHindson** [comment]
Just noticed that the release note is in the single crystal folder for this PR. If you have a minute today please could you move to `docs\source\release\v6.7.0\Diffraction\Powder`

**Comment by RichardWaiteSTFC** [comment]
> Just noticed that the release note is in the single crystal folder for this PR. If you have a minute today please could you move to `docs\source\release\v6.7.0\Diffraction\Powder`

Good spot thanks! I was on auto-pilot!

**Comment by DannyHindson** [comment]
Is the old release note still there in the Single Crystal folder?

**Comment by RichardWaiteSTFC** [comment]
> Is the old release note still there in the Single Crystal folder?

Not since my force-push, sorry!

**Review by DannyHindson — APPROVED** [review]

**Pull request merged**